plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -110,14 +110,34 @@ public CompletionStage<GetDataObjectResponse> getDataObjectAsync(GetDataObjectRe
         return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<GetDataObjectResponse>) () -> {
             try {
                 final GetItemResponse getItemResponse = dynamoDbClient.getItem(getItemRequest);
+                String source;
+                boolean found;
                 if (getItemResponse == null || getItemResponse.item() == null || getItemResponse.item().isEmpty()) {
-                    return new GetDataObjectResponse.Builder().id(request.id()).parser(Optional.empty()).build();
+                    found = false;
+                    source = null;
+                } else {
+                    found = true;
+                    source = getItemResponse.item().get(SOURCE).s();
                 }
-
-                String source = getItemResponse.item().get(SOURCE).s();
+                String simulatedGetResponse = "{\"_index\":\""
+                    + request.index()
+                    + "\",\"_id\":\""
+                    + request.id()
+                    + "\",\"_version\":1,\"_seq_no\":-2,\"_primary_term\":0,\"found\":"
```

Comment:
Are all these values constant for every cases? Like version will be 1 always?

Reply:
Probably need @arjunkumargiri to chime in here; this falls into the same category as "shard" information in DDB which is irrelevant, but I haven't been interacting with the DDB implementation enough to know what (if any) is relevant.

Otherwise best I can do is add a TODO to revisit this later.

Reply:
This would be a concern if plugins depends upon versions for any operation. It is preferred if the SDKClient interface hides all opensearch specific details and provides a generic CRUD+search interface.

Reply:
> It is preferred if the SDKClient interface hides all opensearch specific details

What do we do with OpenSearch customers who are expecting this information and may already have automation relying on the GetResponse API behavior that has been there for years?

I need a really strong argument for taking away information here.

Reply:
> Are all these values constant for every cases? Like version will be 1 always?

Looking more deeply I can just omit these from the strings and it will go with the defaults, which match what was there except version is -1.